src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -103,6 +107,19 @@
     private final Cache<String, ActiveDirectoryGroupDetails> groupCache;
 
     /**
+     * The threadPool to update the cache on background
+     */
+    private final ExecutorService threadPoolExecutor =
+            new ThreadPoolExecutor(
+                    corePoolSize,
+                    maxPoolSize,
+                    keepAliveTime,
+                    TimeUnit.MILLISECONDS,
+                    new ArrayBlockingQueue<Runnable>(queueSize),
+                    new ThreadPoolExecutor.DiscardPolicy()
+            );
```

Comment:
Use a daemon thread factory and a named thread factory so that:

1. This thread pool does not block shutdown
2. The threads in this thread pool can be easily identified in a thread dump